### PR TITLE
fix: add validator information to email/slack alerts

### PR DESCRIPTION
**Pull request by JasonD28**
### SUMMARY

This PR updates alert messages by adding the reason why a SQL-based alert was triggered. Information is taken from the alert's validator string.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
Email:
Screen Shot 2020-09-02 at 12 19 34 PM
Slack:
Screen Shot 2020-09-02 at 12 20 10 PM

### TEST PLAN

- [x] local
- [x] unit test

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/models/alerts.py [hunk 1]
```diff
@@ -201,3 +202,15 @@ def alert(self) -> RelationshipProperty:
             foreign_keys=[self.alert_id],
             backref=backref("validators", cascade="all, delete-orphan"),
         )
+
+    def __str__(self) -> str:
```
Comment (bkyryliuk) [inline superset/models/alerts.py:206]: s/__str__/pretty_print

File: superset/tasks/schedules.py [hunk 2]
```diff
@@ -571,29 +572,31 @@ def deliver_alert(
...
     # Set all the values for the alert report
     # Alternate values are used in the case of a test alert
-    # where an alert has no observations yet
+    # where an alert might not have a validator
     recipients = recipients or alert.recipients
     slack_channel = slack_channel or alert.slack_channel
-    sql = alert.sql_observer[0].sql if alert.sql_observer else ""
-    observation_value = (
-        str(alert.observations[-1].value) if alert.observations else "Value"
+    validation_str = (
```
Comment (bkyryliuk) [inline superset/tasks/schedules.py:578]: s/validation_str/ validation_error_message

File: superset/templates/email/alert.txt [hunk 3]
```diff
@@ -20,6 +20,7 @@
 <p><b>SQL Statement:</b></p>
 <code><mark style="background-color: LightGrey; font-size: 1.1em">{{sql}}</mark></code></p>
 <p><b>SQL Result</b>: {{observation_value}}</p>
+<p><b>Reason For Alert</b>: {{validation_str}}</p>
```
Comment (bkyryliuk) [inline superset/templates/email/alert.txt:23]: let's change it here a bit:
```
Query:
Result:
Reason:
```
to keep it a bit simpler

**Review by bkyryliuk — APPROVED** [review]
LG%nits

**Review by willbarrett — APPROVED** [review]
LGTM once CI passes

**Pull request merged**